[2023-10-22T20:04:03.741+0300] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: dm1trybu.init scheduled__2023-10-21T09:45:00+00:00 [queued]>
[2023-10-22T20:04:03.746+0300] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: dm1trybu.init scheduled__2023-10-21T09:45:00+00:00 [queued]>
[2023-10-22T20:04:03.746+0300] {taskinstance.py:1359} INFO - Starting attempt 1 of 4
[2023-10-22T20:04:03.754+0300] {taskinstance.py:1380} INFO - Executing <Task(PythonOperator): init> on 2023-10-21 09:45:00+00:00
[2023-10-22T20:04:03.757+0300] {standard_task_runner.py:57} INFO - Started process 204877 to run task
[2023-10-22T20:04:03.760+0300] {standard_task_runner.py:84} INFO - Running: ['airflow', 'tasks', 'run', 'dm1trybu', 'init', 'scheduled__2023-10-21T09:45:00+00:00', '--job-id', '95', '--raw', '--subdir', 'DAGS_FOLDER/final_dag_3_models.py', '--cfg-path', '/tmp/tmpz8x7l4p1']
[2023-10-22T20:04:03.762+0300] {standard_task_runner.py:85} INFO - Job 95: Subtask init
[2023-10-22T20:04:03.806+0300] {task_command.py:415} INFO - Running <TaskInstance: dm1trybu.init scheduled__2023-10-21T09:45:00+00:00 [running]> on host dsbulgako1-QEMU-Virtual-Machine
[2023-10-22T20:04:03.879+0300] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_EMAIL='dmitrii.bulghakov@bk.ru' AIRFLOW_CTX_DAG_OWNER='Dmitry Bulgakov' AIRFLOW_CTX_DAG_ID='dm1trybu' AIRFLOW_CTX_TASK_ID='init' AIRFLOW_CTX_EXECUTION_DATE='2023-10-21T09:45:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2023-10-21T09:45:00+00:00'
[2023-10-22T20:04:03.879+0300] {final_dag_3_models.py:71} INFO - Initializing dag and collecting meta-data...
[2023-10-22T20:04:04.175+0300] {migration.py:213} INFO - Context impl PostgresqlImpl.
[2023-10-22T20:04:04.175+0300] {migration.py:216} INFO - Will assume transactional DDL.
[2023-10-22T20:04:04.192+0300] {final_dag_3_models.py:82} INFO - Trying to set new experiment...
[2023-10-22T20:04:04.196+0300] {final_dag_3_models.py:84} INFO - Experiment 'dm1trybu' created successfully!
[2023-10-22T20:04:04.196+0300] {final_dag_3_models.py:90} INFO - Creating nested run in MLFlow.
[2023-10-22T20:04:04.272+0300] {python.py:194} INFO - Done. Returned value was: {'start_tiemstamp': '20231022 20:04', 'run_name': 'parent_standard_3_models', 'experiment_name': 'dm1trybu', 'experiment_id': '8', 'parent_run_id': <ActiveRun: >}
[2023-10-22T20:04:04.283+0300] {xcom.py:661} ERROR - Object of type ActiveRun is not JSON serializable. If you are using pickle instead of JSON for XCom, then you need to enable pickle support for XCom in your airflow config or make sure to decorate your object with attr.
[2023-10-22T20:04:04.283+0300] {taskinstance.py:1935} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/dsbulgako1/my_repos/mlops_project/env/lib/python3.10/site-packages/airflow/utils/json.py", line 91, in default
    return serialize(o)
  File "/home/dsbulgako1/my_repos/mlops_project/env/lib/python3.10/site-packages/airflow/serialization/serde.py", line 178, in serialize
    raise TypeError(f"cannot serialize object of type {cls}")
TypeError: cannot serialize object of type <class 'mlflow.tracking.fluent.ActiveRun'>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/dsbulgako1/my_repos/mlops_project/env/lib/python3.10/site-packages/airflow/utils/session.py", line 74, in wrapper
    return func(*args, **kwargs)
  File "/home/dsbulgako1/my_repos/mlops_project/env/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 2477, in xcom_push
    XCom.set(
  File "/home/dsbulgako1/my_repos/mlops_project/env/lib/python3.10/site-packages/airflow/utils/session.py", line 74, in wrapper
    return func(*args, **kwargs)
  File "/home/dsbulgako1/my_repos/mlops_project/env/lib/python3.10/site-packages/airflow/models/xcom.py", line 244, in set
    value = cls.serialize_value(
  File "/home/dsbulgako1/my_repos/mlops_project/env/lib/python3.10/site-packages/airflow/models/xcom.py", line 659, in serialize_value
    return json.dumps(value, cls=XComEncoder).encode("UTF-8")
  File "/usr/lib/python3.10/json/__init__.py", line 238, in dumps
    **kw).encode(obj)
  File "/home/dsbulgako1/my_repos/mlops_project/env/lib/python3.10/site-packages/airflow/utils/json.py", line 104, in encode
    return super().encode(o)
  File "/usr/lib/python3.10/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/usr/lib/python3.10/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "/home/dsbulgako1/my_repos/mlops_project/env/lib/python3.10/site-packages/airflow/utils/json.py", line 93, in default
    return super().default(o)
  File "/usr/lib/python3.10/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type ActiveRun is not JSON serializable
[2023-10-22T20:04:04.300+0300] {taskinstance.py:1398} INFO - Marking task as UP_FOR_RETRY. dag_id=dm1trybu, task_id=init, execution_date=20231021T094500, start_date=20231022T170403, end_date=20231022T170404
[2023-10-22T20:04:04.313+0300] {standard_task_runner.py:104} ERROR - Failed to execute job 95 for task init (Object of type ActiveRun is not JSON serializable; 204877)
[2023-10-22T20:04:04.336+0300] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2023-10-22T20:04:04.361+0300] {taskinstance.py:2776} INFO - 0 downstream tasks scheduled from follow-on schedule check
